pkg/deploy/api/types.go

```diff
@@ -241,7 +241,11 @@ const (
 	DeploymentStatusReasonAnnotation = "openshift.io/deployment.status-reason"
 	// DeploymentCancelledAnnotation indicates that the deployment has been cancelled
 	// The annotation value does not matter and its mere presence indicates cancellation
+	// DEPRECATED in favor of openshift.io/deployment.cancelled-at
 	DeploymentCancelledAnnotation = "openshift.io/deployment.cancelled"
+	// DeploymentCancelledAtAnnotation indicates that the deployment has been cancelled
+	// The annotation value holds the time a deployment was cancelled in UTC format.
+	DeploymentCancelledAtAnnotation = "openshift.io/deployment.cancelled-at"
```

Comment:
Right - eventually we will have these in RC Conditions.

Upstream issue for adding RC conditions: https://github.com/kubernetes/kubernetes/issues/32863